src/system_simulation/trace_visualization.rb

```diff
@@ -0,0 +1,126 @@
+# Simulate data samples of a system simulation by plotting a trace consisting of transparent circles into the scene.
+class TraceVisualization
+  def initialize
+    # Simulation data to visualize
+    @simulation_data = nil
+
+    # Group containing trace circles.
+    @group = Sketchup.active_model.active_entities.add_group
+
+    # List of trace circles.
+    @trace_points = []
+
+    # Visualization parameters
+    @color = Sketchup::Color.new(72,209,204)
+    @alpha = 0.4
+  end
+
+  def add_trace(node_ids, sparce_factor, data)
+    @simulation_data = data
+    add_circle_trace(node_ids, sparce_factor)
+  end
+
+  def reset_trace()
+    @group = Sketchup.active_model.entities.add_group if @group.deleted?
+    Sketchup.active_model.active_entities.erase_entities(@group.entities.to_a)
+    if @trace_points.count > 0
+      Sketchup.active_model.active_entities.erase_entities(@trace_points)
+    end
+    @trace_points = []
+  end
+
+  private
+
+  def add_circle_trace(node_ids, sparse_factor)
+    @simulation_data.each_with_index do |current_data_sample, index|
+      # thin out points in trace
+      next unless index % sparse_factor == 0
+
+      @group = Sketchup.active_model.entities.add_group if @group.deleted?
+      entities = @group.entities
+
+      materialToSet = Sketchup.active_model.materials.add("MyColor_1")
+      materialToSet.color = @color
+      materialToSet.alpha = @alpha
+
+      edgearray = entities.add_circle(current_data_sample.position_data[node_ids[0]], Geom::Vector3d.new(1,0,0), 1, 10)
+      edgearray.each{|e| e.hidden=true }
+      first_edge = edgearray[0]
+      arccurve = first_edge.curve
+      face = entities.add_face(arccurve)
+      face.material = materialToSet unless face == nil
+
+      edgearray = entities.add_circle(current_data_sample.position_data[node_ids[1]], Geom::Vector3d.new(1,0,0), 1, 10)
+      edgearray.each{|e| e.hidden=true }
+      first_edge = edgearray[1]
+      arccurve = first_edge.curve
+      face = entities.add_face(arccurve)
+      face.material = materialToSet unless face == nil
```

Comment:
This could be very good deduplicated

Reply:
Yep but I decided it's not worth the effort since only on of the methods is used anyways and the others will probably be removed soon...

Update: just removed them...